### PR TITLE
"tx not found" and "block not found" error pages

### PR DESCRIPTION
**Pull request by maxgrapps**
https://github.com/poanetwork/blockscout/issues/1747 - issue

1) Add _errors.scss to css/compontents folder
2) Connect _errors.scss at the bottom of app.scss
3) Add "errors-img" folder and "errors-tpl" folder to static/images

NOTE: This pages made for ethereum mainnet theme. Only difference for another themes is in images. I'll upload in in another pull requests.

screencapture-localhost-4000-tx-0x32e2e16e6a06a1e67cd023a8a68c54a5d590f9426a6fb672041410dabad31e96-2019-05-21-19_19_09

screencapture-localhost-4000-blocks-24-transactions-2019-05-21-19_18_50

### CODE REVIEW

File: apps/block_scout_web/assets/static/images/errors-tpl/tx-not-found.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+				<div class="tx-nf-illustr">
+					<img alt="Block Not Found" src="../errors-img/eth-tx-not-found.png" srcset="../eth-tx-not-found@2x.png 2x">
+				</div>
+				<div class="tx-nf-content">
+					<h1 class="card-title error-title">Sorry, We are unable to locate this transaction Hash</h1>
+					<div class="tx-nf-blocks">
+						<div class="tx-nf-blocks-row">
+							<div class="tx-nf-block">
+								<span class="tx-nf-number">1</span>
+								<p class="error-descr">If you have just submitted this transaction please wait for at least  30 seconds before refreshing this page.</p>
```
Comment (vbaranov) [inline apps/block_scout_web/assets/static/images/errors-tpl/tx-not-found.html]: double whitespace. Please remove.

File: apps/block_scout_web/assets/static/images/errors-tpl/tx-not-found.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+							</div>
+						</div>
+						<div class="tx-nf-blocks-row">
+							<div class="tx-nf-block">
+								<span class="tx-nf-number">3</span>
+								<p class="error-descr">During times when the network is busy (i.e during ICOs) it can take a while for your transaction to propagate through the network and for us to index it.</p>
+							</div>
+							<div class="tx-nf-block">
+								<span class="tx-nf-number">4</span>
+								<p class="error-descr">If it still does not show up after 1 hour, please check with your sender/exchange/wallet/transaction provider for additional & information.</p>
```
Comment (vbaranov) [inline apps/block_scout_web/assets/static/images/errors-tpl/tx-not-found.html]: Please remove ampersand.

**Comment by maxgrapps** [comment]
@vbaranov amp and double spacing are removed.
Also, I added html code to eex templates. Hope it's working correct.
Can you send me an html link how to view this errors, not only a link to a file? Or guide what I should do to see an errors?

**Comment by vbaranov** [comment]
@maxgrapps I thought I put those links to the review. But, Indeed, they are missing.
Here are the links, you can try in local deployment to see the both errors:
1. "block not found page": `./blocks/24`
2. "transaction not found page": `./tx/0x32e2e16e6a06a1e67cd023a8a68c54a5d590f9426a6fb672041410dabad31e96`

**Comment by maxgrapps** [comment]
@vbaranov Seems, it should work correct now. Can you check the code please?

**Comment by vbaranov** [comment]
@maxgrapps 
`gettext` test failed
steps to fix: https://forum.poa.network/t/faq-fixing-gettext-error-in-blockscout/2444

And please attach screenshots of the pages from your local launch.

**Comment by vbaranov** [comment]
@maxgrapps 

Also, remove obsolete assets:
`apps/block_scout_web/assets/static/images/errors-tpl/block-not-found.html`
`apps/block_scout_web/assets/static/images/errors-tpl/tx-not-found.html`

**Comment by maxgrapps** [comment]
@vbaranov Seems, now it's fine. Screenshots also were reattached.

File: apps/block_scout_web/lib/block_scout_web/templates/block_transaction/404.html.eex [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
-            <%= block_not_found_message(@block_above_tip) %>
-          </div>
-        </div>
-      </div>
+  <div class="block-not-found">
+    <div class="block-not-found-img">
+      <img alt="Block Not Found" src="/images/errors-img/eth-block-not-found.png" srcset="/images/eth-block-not-found@2x.png 2x">
+    </div>
+    <div class="block-not-found-content">
+      <h1 class="card-title error-title">Lorem Ipsum Dolor</h1>
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/templates/block_transaction/404.html.eex]: It shouldn't be "Lorem Ipsum Dolor" here. We should substitute here `<%= gettext("Block Details") %>`

File: apps/block_scout_web/lib/block_scout_web/templates/block_transaction/404.html.eex [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
+      <p class="error-descr">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua</p>
```
Comment (vbaranov) [inline apps/block_scout_web/lib/block_scout_web/templates/block_transaction/404.html.eex]: The same as above, replace with `<%= block_not_found_message(@block_above_tip) %>`

**Comment by maxgrapps** [comment]
@vbaranov 
- I added content instead of lorem ipsum for "block not found" page
- What content should I keep for "transaction not found" page? Current or content from mockups?
- Executed mix test also.

**Comment by vbaranov** [comment]
@maxgrapps tests are failed. You should fix them manually going through errors from `mix test`, that related to updating of error pages.
`gettext` test is again failed.

>What content should I keep for "transaction not found" page? Current or content from mockups?

Yes, from mockups

**Comment by maxgrapps** [comment]
@vbaranov Do you mean this warning and error below in wallaby.ex? 
Screen Shot 2019-05-22 at 14 59 11

**Comment by vbaranov** [comment]
@maxgrapps 

`mix test --exclude no_geth`

```
1) test GET index/2 non-consensus block number above consensus block number is treated as consensus number above tip (BlockScoutWeb.BlockTransactionControllerTest)
     test/block_scout_web/controllers/block_transaction_controller_test.exs:59
     Assertion with == failed
     code:  assert conn |> html_response(404) |> Floki.find(~S'[data-selector="block-not-found-message"') |> Floki.text() |> String.trim() == "Easy Cowboy! This block does not exist yet!"
     left:  ""
     right: "Easy Cowboy! This block does not exist yet!"
     stacktrace:
       test/block_scout_web/controllers/block_transaction_controller_test.exs:70: (test)

........

  2) test GET index/2 non-consensus block number without consensus blocks is treated as consensus number above tip (BlockScoutWeb.BlockTransactionControllerTest)
     test/block_scout_web/controllers/block_transaction_controller_test.exs:48
     Assertion with == failed
     code:  assert conn |> html_response(404) |> Floki.find(~S'[data-selector="block-not-found-message"') |> Floki.text() |> String.trim() == "Easy Cowboy! This block does not exist yet!"
     left:  ""
     right: "Easy Cowboy! This block does not exist yet!"
     stacktrace:
       test/block_scout_web/controllers/block_transaction_controller_test.exs:56: (test)

```

**Comment by maxgrapps** [comment]
@vbaranov Okay. How can I fix this? I don't understand why it happens? I just remove old page code and add new html with some your template code

**Comment by vbaranov** [comment]
@maxgrapps if you are unable to fix this, close your PR and open it to another branch https://github.com/poanetwork/blockscout/tree/errors
Thanks

**Pull request closed without merging**